presto-spi/src/main/java/com/facebook/presto/spi/function/AccumulatorStateFactory.java

```diff
@@ -25,4 +25,16 @@
     T createGroupedState();
 
     Class<? extends T> getGroupedStateClass();
+
```

Comment:
i want to understand this, in-person chat maybe. looks interesting....

the benefit of a dedicated scratch class vs being forced to use the real agg class could lead to optimizations (i wouldn't call it a scratch class, but base class, since it's the base empty class that gets base.combine(firstData)

was that the thinking?